xinetd: Add xinetd package

terasawa-lin opened the pull request:
Modifications to the recipe
add:
inherit gettext for configure
Run autogen.sh on do_configure
Added parameters to oe_runconf
do_compile
RDEPEND

remove:
SRC_URI
S

change:
reference directory for do_install

# Purpose of pull request

This pr adds xinetd package based on poky/meta/recipes-extended/xinetd/xinetd_2.3.15.bb

# Test
## How to install
Add the following to conf/local.conf.
```
IMAGE_INSTALL_append = " xinetd"
```

## How to test
Run following command from console

```
# vi /etc/xinetd.d/echo
disable = yes
↓
disable = no

# /etc/init.d/xinetd restart
```

## Test result
If you connect to the port number 7 of the local host with the nc command and enter a character, the entered character is returned.
```
# nc localhost 7
aaa
aaa
^C
```

hiraku-wfs commented on recipes-debian/xinetd/xinetd_debian.bb:
```diff
@@ -0,0 +1,87 @@
+PR = "r2"
+
+# Blacklist a bogus tag in upstream check
+UPSTREAM_CHECK_GITTAGREGEX = "xinetd-(?P<pver>(?!20030122).+)"
```
We use Debian source package, so we don't need this.

hiraku-wfs commented on recipes-debian/xinetd/xinetd_debian.bb:
```diff
@@ -0,0 +1,87 @@
+
+FILESPATH_append = ":${COREBASE}/meta/recipes-extended/xinetd/xinetd"
+
+UPSTREAM_VERSION_UNKNOWN = "1"
```
Is this necessary?

hiraku-wfs commented on recipes-debian/xinetd/xinetd_debian.bb:
```diff
@@ -0,0 +1,87 @@
+
+UPSTREAM_VERSION_UNKNOWN = "1"
+
+SRCREV = "68bb9ab9e9f214ad8a2322f28ac1d6733e70bc24"
```
We use Debian source package, so we don't need this.

terasawa-lin commented:
Fixed for 3 comments.

masami256 commented on recipes-debian/xinetd/xinetd_debian.bb:
```diff
@@ -0,0 +1,77 @@
+DEBIAN_PATCH_TYPE = "quilt"
+DEBIAN_QUILT_PATCHES = ""
+
+S = "${WORKDIR}/${BP}"
```
```DEBIAN_UNPACK_DIR``` is same as ``` "${WORKDIR}/${BP}"``` and ```S``` will be set  as ```DEBIAN_UNPACK_DIR```. so you don't need to set S here.

terasawa-lin committed:
xinetd: Add xinetd package

Modifications to the recipe
add:
inherit gettext for configure
Run autogen.sh on do_configure
Added parameters to oe_runconf
do_compile
RDEPEND

remove:
PR
UPSTREAM_CHECK_GITTAGREGEX
UPSTREAM_VERSION_UNKNOWN
SRCREV
SRC_URI
S

change:
reference directory for do_install

terasawa-lin commented:
変数Sを取り除きました。

terasawa-lin commented:
Removed the variable S.

masami256 commented:
Confirmed that xinet.d works both sysvinit and systemd fine.
